packages/beacon-node/src/chain/stateCache/stateContextCache.ts

```diff
@@ -84,6 +84,33 @@ export class StateContextCache {
     return this.cache.size;
   }
 
+  updateUnfinalizedPubkeys(validators: UnfinalizedPubkeyIndexMap): void {
+    let totalNumStatesUpdated = 0;
+
+    const addTimer = this.metrics?.addPubkeyTime.startTimer();
```

Comment:
I am aware that we use this pattern everywhere, but are this functions expected to throw? If they do the timer value won't be much relevant, I would just time without try / catch (add a comment of why) and if it errors then there's no timing information

Reply:
> I am aware that we use this pattern everywhere, but are this functions expected to throw? If they do the timer value won't be much relevant, I would just time without try / catch (add a comment of why) and if it errors then there's no timing information

`addFinalizedPubkeys()` is expected to throw error if it is pre-6110. But I agree in the case an error is thrown, the timer value will be irrelevant.